Source/Core/Core/IOS/IOSC.cpp

```diff
@@ -0,0 +1,246 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <algorithm>
+#include <cstring>
+
+#include <mbedtls/sha1.h>
+
+#include "Common/Assert.h"
+#include "Common/ChunkFile.h"
+#include "Common/Crypto/AES.h"
+#include "Common/Crypto/ec.h"
+#include "Core/IOS/Device.h"
+#include "Core/IOS/IOSC.h"
+#include "Core/ec_wii.h"
+
+namespace IOS
+{
+namespace HLE
+{
+IOSC::IOSC()
+{
+  LoadDefaultEntries();
+}
+
+IOSC::~IOSC() = default;
+
+ReturnCode IOSC::CreateObject(Handle* handle, ObjectType type, ObjectSubType subtype, u32 pid)
+{
+  auto iterator = FindFreeEntry();
+  if (iterator == m_key_entries.end())
+    return IOSC_FAIL_ALLOC;
+
+  iterator->in_use = true;
+  iterator->type = type;
+  iterator->subtype = subtype;
+  iterator->owner_mask = 1 << pid;
+
+  *handle = GetHandleFromIterator(iterator);
+  return IPC_SUCCESS;
+}
+
+ReturnCode IOSC::DeleteObject(Handle handle, u32 pid)
+{
+  if (handle <= 0x11 || !HasOwnership(handle, pid))
```

Comment:
Is ```0x11``` referring to the DefaultHandle count? Shouldn't it be 11 rather than 17 (or the enum values might be wrong)? You might consider to put an entry like ```DEFAULT_HANDLE_COUNT```, if so.